### PR TITLE
[LE-71] Event payload only contains string with the name of the event, not an object

### PR DESCRIPTION
**Pull request by lorexsan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,7 +81,7 @@
     "enzyme": "3.3.0",
     "enzyme-adapter-react-16": "1.1.1",
     "enzyme-to-json": "3.1.2",
-    "eslint": "3.19.0",
+    "eslint": "5.16.0",
```
Comment (dfournier) [inline package.json]: As we discussed, it's impossible you update a dependency without changing `package-lock.json`. If something is not clear about this point, feel free to ping me.

File: src/helpers/hub.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 export type Listener = (type: string, payload: any) => void;
```
Comment (dfournier) [inline src/helpers/hub.ts]: `Listener` doesn't look like to be used anymore. I suggest removing it.

File: src/helpers/hub.ts [hunk 3]
```diff
@@ -28,7 +29,7 @@ export default class Hub {
 
     const listener = this.listeners[type];
     if (listener) {
-      listener(type, payload);
+      listener(payload);
```
Comment (dfournier) [inline src/helpers/hub.ts]: This is a huge breaking change. Why don't we reply to the person who raised the issue the object is on the second argument?

File: src/helpers/hub.ts (hunk 2 above)
Comment (nickvdyck) [reply src/helpers/hub.ts]: `Listener` is still used here: https://github.com/datacamp/datacamp-light/blob/33d9487fd6cd66f33b045fff579718585b91067c/src/redux/exercise.ts#L41

File: src/helpers/hub.ts (hunk 3 above)
Comment (nickvdyck) [reply src/helpers/hub.ts]: We could,  but we did not solely make this changed based on a use request. The current implementation is wrong according to the specs: https://cdn.datacamp.com/dcl-react-prod/index.html#tutorial-customizing-integrating. 

So if we don't want to make this breaking change we have 2 options.
- Update the docs so they reflect reality, although it doesn't make any sense to me to pass the event as the first argument
- make the change but do it in a way that is backwards compatible for now and rectify this later when we release a new major version. We can easily check how many arguments are getting passed and jump into a different code based on that.

File: package.json (hunk 1 above)
Comment (nickvdyck) [reply package.json]: Why do we need to update our linter?

Comment (lorexsan) [reply package.json]: No, this shouldn't be part of this pull request, my bad. This has already been fixed with the help of David on Friday, doesn't need to change now. Can we remove this change?

File: src/helpers/hub.ts (hunk 3 above)
Comment (dfournier) [reply src/helpers/hub.ts]: I completely agree with you it doesn't make sense to pass the event type as the first argument. However, people using that feature probably found out the payload is passed on the second argument. This was suggested [here](https://github.com/datacamp/datacamp-light/issues/103#issuecomment-415328094) and looks like the user did it according to his [reply](https://github.com/datacamp/datacamp-light/issues/103#issuecomment-415418844).

With no versioning in place, I suggest keeping this fix for version 3.

Comment (dfournier) [reply src/helpers/hub.ts]: And as you suggested, update the documentation 😄

File: src/helpers/hub.ts (hunk 2 above)
Comment (dfournier) [reply src/helpers/hub.ts]: Why not move it then? Are they really coupled? It's not that obvious in `datacamp-light/src/redux/exercise.ts`

**Pull request merged**